agimus_controller/croco_hpp.py

```diff
@@ -314,13 +341,13 @@ def compute_next_step(self, x, problem):
         m.calc(d, x, self.prob.solver.us[0])
         return d.xnext.copy()
 
-    def do_mpc(self, path_terminal_idx, T, node_idx_breakpoint=None):
-        self.prob.set_models([path_terminal_idx])
+    def do_mpc(self, T, node_idx_breakpoint=None):
+        self.prob.set_models()
         self.prob.create_whole_problem()
-        self.prob.set_xplan_and_uref(0, path_terminal_idx)
+        # self.prob.set_xplan_and_uref(0, path_terminal_idx)
 
         problem = self.prob.create_problem(T)
```

Comment:
this is a mistake here. You have a python module named `problem.py`
it's dangerous to call a variable `problem`.